Add kubectl overview page

hw-qiaolei opened the pull request:
This is an effort try to fix: #11814

@bgrant0607 @hurf

hw-qiaolei committed:
Add kubectl overview page

bgrant0607 commented:
Thanks!

cc @kubernetes/kubectl

bgrant0607 commented:
@RichieEscarez Please suggest where links to this should be added, or existing links to the generated docs should be changed to point here.

jlowdermilk commented on docs/user-guide/kubectl-overview.md:
```diff
@@ -0,0 +1,189 @@
+
+## Common Operations
+
+For explanation, here I gave some mostly often used `kubectl` command examples. Please replace sample names with actually values if you would like to try these commands.
```
s/actually/actual

jlowdermilk commented:
This is a good intro page. My only concern is that it will not stay up to date unless it's autogenerated. A good solution would be to modify [gen_kubectl_docs.go](https://github.com/kubernetes/kubernetes/blob/master/cmd/gendocs/gen_kubectl_docs.go) to generate this doc, if you're interested in doing that. In particular the examples, operations, and resource types should be retrieved from the built kubectl. I'm also fine with submitting this as is (once appropriate links to/from existing kubectl docs are updated) and putting a note in #14177 to autogenerate this file.

JanetKuo commented:
Agree on auto-generate this doc. If we're submitting this as is please add `kubectl edit` too. https://github.com/kubernetes/kubernetes/blob/master/docs/user-guide/kubectl/kubectl_edit.md

hw-qiaolei commented:
@jlowdermilk thank you for reviewing this and giving these good suggestions. Yes this page is manually written and will be out of date if there is a change of `kubectl`. I will look into how to auto-generate it.
My initial thought is that this page should include some most often used examples(not all of the commands), and for readability I rewrote these examples(not all of the examples given by a certain command, and changed some variables, explanations etc). They should be **small and stable, more user-friendly**.
Anyway, I will look into how the auto-generate it and try to keep a balance between **keeping it up with `kubectl` and being small and user-friendly**.

hw-qiaolei commented:
@JanetKuo thank you for the comment. I have added `kubectl edit` manually and will look into how to auto-generate this doc.

hw-qiaolei commented:
cc @alfred-huangjian

RichieEscarez commented:
Sorry for the delay. Starting my review now.

RichieEscarez commented:
After a discussion with @jlowdermilk, im going to LGTM this to make it live and then open a separate PR for my edits.
